mmdet/models/anchor_heads/fovea_head.py

```diff
@@ -223,6 +223,7 @@ def loss(self,
                 pos_bbox_targets,
                 pos_weights,
                 avg_factor=num_pos)
+            loss_bbox = loss_bbox.unsqueeze(0)
```

Comment:
Unsqueezing the `loss_bbox` when `num_pos>0` is a little bit risky since it is more commonly used. It seems that other detectors' losses also do not have such a pattern.
So maybe it is better and safer to squeeze the loss_bbox when `num_pos==0`?

Reply:
Squeezing when num_pos==0 would make loss a scalar, which sometimes causes pytorch to throw a warning "UserWarning: Was asked to gather along dimension 0, but all input tensors were scalars; will instead unsqueeze and return a vector."
Otherwise, I do not see why the author chose:
`loss_bbox = torch.tensor([0],
                                     dtype=flatten_bbox_preds.dtype,
                                     device=flatten_bbox_preds.device)`

and not
`loss_bbox = torch.tensor(0,
                                     dtype=flatten_bbox_preds.dtype,
                                     device=flatten_bbox_preds.device)`

Reply:
However, in the codebases, the final losses are always a scalar. 
So we suggest changing the case`num_pos==0` and use either
```python
loss_bbox = torch.tensor([0], dtype=flatten_bbox_preds.dtype, device=flatten_bbox_preds.device).sum()
```
or 
```python
loss_bbox = torch.tensor(0, dtype=flatten_bbox_preds.dtype, device=flatten_bbox_preds.device)
```
Does this work for you?

Reply:
Sure, I will do that.

Reply:
Thanks.